### PR TITLE
feat(test-runner): introduce test.step

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/test-api/class-test.md [hunk 1]
```diff
@@ -709,6 +709,41 @@ Optional condition - either a boolean value, or a function that takes a fixtures
...
+
+test('test', async ({ page }) => {
+  await test.step('Log in', async () => {
+    // ...
+  });
+});
+```
+
+### param: Test.step.title
+- `title` <[int]>
```
Comment (dgozman) [inline docs/src/test-api/class-test.md:737]: [string]

File: tests/playwright-test/reporter.spec.ts [hunk 2]
```diff
@@ -274,3 +278,109 @@ test('should report api steps', async ({ runInlineTest }) => {
...
+    `%% end {\"title\":\"expect.toBe\",\"category\":\"expect\",\"error\":{\"message\":\"expect(received).toBe(expected) // Object.is equality\\n\\nExpected: 2\\nReceived: 1\",\"stack\":\"<stack>\"}}`,
+    `%% end {\"title\":\"First step\",\"category\":\"test.step\",\"error\":{\"message\":\"expect(received).toBe(expected) // Object.is equality\\n\\nExpected: 2\\nReceived: 1\",\"stack\":\"<stack>\"}}`,
+    `%% begin {\"title\":\"After Hooks\",\"category\":\"hook\"}`,
+    `%% begin {\"title\":\"browserContext.close\",\"category\":\"pw:api\"}`,
+    `%% end {\"title\":\"browserContext.close\",\"category\":\"pw:api\"}`,
+    `%% end {\"title\":\"After Hooks\",\"category\":\"hook\"}`,
+  ]);
+});
+
+function stripEscapedAscii(str: string) {
```
Comment (dgozman) [inline tests/playwright-test/reporter.spec.ts:384]: I think there is already such a function somewhere next to fixtures.

File: types/test.d.ts [hunk 3]
```diff
@@ -2093,6 +2093,33 @@ export interface TestType<TestArgs extends KeyValue, WorkerArgs extends KeyValue
...
+   *   await test.step('Log in', async () => {
+   *     // ...
+   *   });
+   * });
+   * ```
+   *
+   * @param title Step name.
+   * @param body Step body.
+   */
+  step(title: string, body: () => Promise<void>): Promise<void>;
```
Comment (dgozman) [inline types/test.d.ts:2122]: Let's make body return `Promise<any>` to not be too restrictive.

**Review by dgozman — APPROVED** [review]

File: tests/playwright-test/reporter.spec.ts (hunk 2 above)
Comment (pavelfeldman) [reply tests/playwright-test/reporter.spec.ts:384]: This is escaping JSON-stringified, already-escaped sequence.

**Pull request merged**